### ISSUE TITLE
LunarG/MoltenVK glfwGetRequiredInstanceExtensions returns NULL

### ISSUE THREAD

**Issue by mokafolio** [glfw/glfw#1299]
Hi,
I am having trouble getting the vulkan example to work on osx. I installed Lunarg/MoltenVG Vulkan SDK version 1.1.77.0.

The odd thing is, `glfwVulkanSupported` returns true, only when I try to setup the device using `glfwGetRequiredInstanceExtensions` I run into trouble as it returns `NULL`. I ended up building GLFW master from source and while the `vulkan.c` example/test compiles properly, I experience the same issue when I run it:

```
glfwGetRequiredInstanceExtensions failed to find the platform surface extensions.

Do you have a compatible Vulkan installable client driver (ICD) installed?
Please look at the Getting Started guide for additional information.
```

Am I missing something? I'd appreciate any ideas/suggestions!

Thank you!

EDIT: I realized I forgot to specify which osx I am on: _10.13.4_

**Comment by elmindreda** [comment]
Seems like GLFW finds the Vulkan loader but the Vulkan loader doesn't find the MoltenVK ICD.

How did you install the SDK? The loader needs to find the JSON file and that file needs to point to the ICD. If you haven't, read about `VK_ICD_FILENAMES` in the LunarG SDK readme.

**Comment by mokafolio** [comment]
I installed LunarG from the official website and simply put it in a place in my user directory and then set then set `export VULKAN_SDK=` to the correct path.

Indeed the `VK_ICD_FILENAMES` was the step I was missing:
`export VK_ICD_FILENAMES=$VULKAN_SDK/etc/vulkan/icd.d/MoltenVK_icd.json`

Fixed it for both, the `vulkan.c` that ships with GLFW and my own test application.

Apologies for being blind and for the quick response!
